src/libs/pois.js

```diff
@@ -0,0 +1,71 @@
+
+import { version, sources } from 'config/constants.yml';
+import ExtendedString from 'src/libs/string';
+import IdunnPoi from 'src/adapters/poi/idunn_poi';
+import LatLonPoi from 'src/adapters/poi/latlon_poi';
+
+// POI from/to url functions
+
+export function toUrl(poi) {
+  if (poi.id === 'geolocalisation' || poi.type === 'latlon') {
+    return `latlon:${poi.latLon.lat.toFixed(5)}:${poi.latLon.lng.toFixed(5)}`;
+  }
+  return `${poi.id}@${ExtendedString.slug(poi.name)}`;
+}
+
+export function toAbsoluteUrl(poi) {
+  const { protocol, host } = window.location;
+  const baseUrl = window.baseUrl;
+  const lat = poi.latLon.lat.toFixed(7);
+  const lon = poi.latLon.lng.toFixed(7);
+  const mapHash = `#map=${getBestZoom(poi.zoom)}/${lat}/${lon}`;
+  return `${protocol}//${host}${baseUrl}place/${toUrl(poi)}/${mapHash}`;
+}
+
+export function fromUrl(urlParam) {
+  if (!urlParam) {
+    return Promise.reject();
+  }
+
+  const latLonUrlRegex = /^latlon:(-?\d*\.\d*):(-?\d*\.\d*)(?:@(.*))?/;
+  let urlData = urlParam.match(latLonUrlRegex);
+  if (urlData) {
+    const [ _, lat, lng, label ] = urlData;
+    const latLng = { lat: parseFloat(lat), lng: parseFloat(lng) };
+    return Promise.resolve(
+      new LatLonPoi(latLng, label ? ExtendedString.htmlEncode(label) : null)
+    );
+  } else {
```

Comment:
Since you return in the `if` block, wouldn't remove the `else` clause better? I have a personal preference over less indented code but up to you!

Reply:
I prefer it too, I'll remove the `else` block.